src/core/utils/url.tsx

```diff
@@ -0,0 +1,49 @@
+import React, { useEffect, useState } from "react";
+import { RootState, store, useSelector } from "../store";
+import { addUrlEntries } from "../url.slice";
+
+export const useUrls = () => {
+  const { data } = useSelector((state: RootState) => state.url);
+  return data;
+};
+
+export const withUrl = <T,>(Component: React.ComponentType<T>) => {
+  return (props: T) => {
+    const [propsWithoutUrl, setPropsWithoutUrl] = useState({});
+
+    useEffect(() => {
+      const pattern = /.*Url$/g;
+      // Match all props that ends with "Text".
+      const urlEntries = Object.fromEntries(
+        Object.entries(props).filter(([prop]) => {
+          return String(prop).match(pattern);
+        })
+      );
+      // and match all props that do NOT end with "Text".
+      const nonUrlEntries = Object.fromEntries(
+        Object.entries(props).filter(([prop]) => {
+          return !String(prop).match(pattern);
+        })
+      );
+      // If we do have props that are not text props
```

Comment:
There seems to be a wee bit of copy/pasting here.

Reply:
you´re right